### DISCUSSION TITLE
Release v0.21.3 Beta

### DISCUSSION THREAD

**Discussion by PrestonN** [FreeTubeApp/FreeTube#5516] [Announcements]
## Summary

Playback is once again having some issues so we are once again releasing an update to fix the issue.

## Fixes
* Updated YouTube.js to 10.3.0 to fix recent issues and switch streams from the iOS client to workaround playback issues by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/5508

**Full Changelog**: https://github.com/FreeTubeApp/FreeTube/compare/v0.21.2-beta...v0.21.3-beta

<hr /><em>This discussion was created from the release <a href='https://github.com/FreeTubeApp/FreeTube/releases/tag/v0.21.3-beta'>Release v0.21.3 Beta</a>.</em>

**Comment by Scary-fox** [comment]
The video player is still broken for me, it at least loads the page now but the video never plays. Error code: (NotSupportedError: Failed to load because no supported source was found.)

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
Did you update to 0.21.3? AND are you using the Local API?

**Reply by Scary-fox** [reply]
Yes.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
is the following setting enabled?

Player settings -> Proxy Videos Through Invidious

If so try to disable it.

**Reply by Scary-fox** [reply]
Nope, it's off.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
Hmm okay. 

Please do following:

1. Open FT
2. Open the console in the dev tools, `ctrl + shift + i`
3. Watch video
4. Make screenshot of all the errors in the console.

**Reply by Scary-fox** [reply]
That doesn't seem to do anything?
Edit: nvm, it's option command instead of control shift on Mac

**Reply by Scary-fox** [reply]
<img width="538" alt="image" src="https://github.com/user-attachments/assets/09dd377e-7a2d-446e-ac01-39a39ed7ab89">

**Reply by PikachuEXE** [reply]
That error seems to be unable to connect to youtube server (`xxxx.googlevideo.com`)
You might want to check your network connection setup?

**Reply by Scary-fox** [reply]
No, my connection is fine, Youtube is just starting to be throttled I guess so I can't seem to use FreeTube anymore, quite sad. Sorry for the time waste I guess.

**Reply by PikachuEXE** [reply]
throttled should be represented by http status code not ERR_TIMED_OUT
Try it later

**Reply by Scary-fox** [reply]
I'm not really knowledgable so I won't doubt you about this stuff, but it certainly matches every other site that I've seen blocked or made near inaccessible before. Always some vague comment about a connection timing out yet if you turn on a vpn it suddenly works which it shouldn't if it was a connection issue.

**Reply by Mike221223** [reply]
I have the same issue, and I don't think it's on the client. YouTube recently started being blocked or inhibited somehow in Russia, where I happen to live. YouTube itself ceased working properly for me. Videos only load properly after several page refreshes.

**Reply by Scary-fox** [reply]
> I have the same issue, and I don't think it's on the client. YouTube recently started being blocked or inhibited somehow in Russia, where I happen to live. YouTube itself ceased working properly for me. Videos only load properly after several page refreshes.

I reccomend you to download Tor and use the inbuilt proxy featue in FreeTube, I did it an it fixed the throttling. There is a bit of a learning curve to it so read the Freetube documentation on Proxying with Tor.

**Reply by Mike221223** [reply]
> > I have the same issue, and I don't think it's on the client. YouTube recently started being blocked or inhibited somehow in Russia, where I happen to live. YouTube itself ceased working properly for me. Videos only load properly after several page refreshes.
> 
> I reccomend you to download Tor and use the inbuilt proxy featue in FreeTube, I did it an it fixed the throttling. There is a bit of a learning curve to it so read the Freetube documentation on Proxying with Tor.

Thank you!

**Comment by Smiths76245** [comment]
Was having lots of glitchy lagging playback issues too but changed back to 'local api' after previous version issues had us mucking about with several General settings to invidious .. Anyway all good now, such a great App. Muchos Amigos

**Comment by mikebronner** [comment]
I noticed with release 0.21.3 when playing back using the local API, the quality seems to always start at the lowest resolution and not improve until I manually select 1080, even if my default is already at 1080.

Thanks for all the other fixes to the local API! :)

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
We have also identified this issue and we are working on resolving it.

**Comment by Aslinjean** [comment]
Every time I install FreeTube on my 2020 M1 Mac, Apple tells me it's dangerous for my device and automatically refuses the application. So, can someone tell me what I need to do to successfully download FreeTube on my device? Please!

**Reply by mikebronner** [reply]
> Every time I install FreeTube on my 2020 M1 Mac, Apple tells me it's dangerous for my device and automatically refuses the application. So, can someone tell me what I need to do to successfully download FreeTube on my device? Please!

I'm assuming you are able to download and install FreeTube, and the warning happens when you try to open it? If that is the case, cancel the message, and go to **System Settings > Privacy & Security**, then scroll all the way to the bottom to the **Security** section, and you should see a prompt there asking if you want to allow FreeTube to run. Once you approve it, it should open as expected.

**Reply by Aslinjean** [reply]
> > Every time I install FreeTube on my 2020 M1 Mac, Apple tells me it's dangerous for my device and automatically refuses the application. So, can someone tell me what I need to do to successfully download FreeTube on my device? Please!
> 
> 
> 
> I'm assuming you are able to download and install FreeTube, and the warning happens when you try to open it? If that is the case, cancel the message, and go to **System Settings > Privacy & Security**, then scroll all the way to the bottom to the **Security** section, and you should see a prompt there asking if you want to allow FreeTube to run. Once you approve it, it should open as expected.

Thank you, it works, but every time I start a video, it gives me a "Local API Error". When I change it and set it to "Invidious API", I STILL ENCOUNTER THE SAME PROBLEM AND I CAN'T SUCCESSFULLY PLAY ANY VIDEO.

**Reply by mikebronner** [reply]
This is what my settings look like:

<img width="1370" alt="Screenshot 2024-08-14 at 05 45 02" src="https://github.com/user-attachments/assets/40a822eb-68e1-4c5f-8664-87c5da1af3cb">

I believe using Invidious is hit-or-miss at the moment, as I cannot get that to work either.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
What Local API Error? Are you sure you are on v0.21.3?
